### PR TITLE
Mock the Census API for tests so that Actions doesn't actually hit the API when running tests

### LINKED ISSUE #861: Mock the Census API for tests so that Actions doesn't actually hit the API when running tests
## Action Requested
Mock the Census API for tests so that Actions doesn't actually hit the API when running tests. If there is any kind of rate limiting or issue with the external API we don't want that to halt our ability to merge PRs.

AS a $USER, I want to mock the Census API in our tests SO THAT unit tests won't fail when Actions run on PRs due to rate limiting or connection issues to an external API

## Acceptance Criteria
When the mocking function passes its tests and there are no additional issues from other tests

## Background & Context
This is important so that when other tickets are completed and you want to merge to `main` you are not stopped because a unit test for code you're not working on is failing because it's actually hitting an external API and erroring out.

### PR DESCRIPTION
**Pull request by robertmitchellv**
# PULL REQUEST

## Summary
Mock the Census API for tests so that Actions doesn't actually hit the API when running tests. If there is any kind of rate limiting or issue with the external API we don't want that to halt our ability to merge PRs.

## Related Issue
Fixes #861 

## Additional Information
This is important so that when other tickets are completed and you want to merge to main you are not stopped because a unit test for code you're not working on is failing because it's actually hitting an external API and erroring out.

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

**Review by m-goggins — APPROVED** [review]
Sooo much cleaner! 🚀

**Comment by robertmitchellv** [comment]
@emmastephenson rightly pointed out that there are also geocoding functions running in the `fhir/geospatial/test_fhir_census_client.py` that also need to be mocked; will work on that now

File: containers/record-linkage/app/utils.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import subprocess
 from typing import Literal
 
+
```
Comment (DanPaseltiner) [inline containers/record-linkage/app/utils.py:7]: Is this extra line a Black thing? No big deal at all, just scratching my head.

**Review by DanPaseltiner — APPROVED** [review]
LGTM! Nice work @robertmitchellv. I'm going to have to read the docs on `@pytest.fixture` and use it the next time I'm writing unit tests. 🚀

File: tests/geospatial/test_census_geocode_client.py [hunk 2]
```diff
@@ -184,7 +165,8 @@ def test_geocode_from_dict():
 
     # Test ambiguous address
     ambiguous_address_dict = {"street": "123 Main Street"}
-    census_client._call_census_api.return_value = ambiguous_address_dict
+    monkeypatch.setattr(census_client, "_call_census_api", mock_ambiguous_address)
+    # census_client._call_census_api.return_value = ambiguous_address_dict
```
Comment (emmastephenson) [inline tests/geospatial/test_census_geocode_client.py:169]: Can you remove this comment?

**Review by emmastephenson — APPROVED** [review]
Looks great! Thanks for figuring out that other test!

File: containers/record-linkage/app/utils.py (hunk 1 above)
Comment (robertmitchellv) [reply containers/record-linkage/app/utils.py:7]: @DanPaseltiner not a black thing; that's just an accidental extra line; going to fix now

File: tests/geospatial/test_census_geocode_client.py (hunk 2 above)
Comment (robertmitchellv) [reply tests/geospatial/test_census_geocode_client.py]: ah good catch! just removed

**Pull request merged**